evm/ts/src/axelar.ts

```diff
@@ -0,0 +1,185 @@
+import { Chain, Network } from "@wormhole-foundation/sdk-base";
+
+// Using direct API calls instead of importing the entire @axelar-network/axelarjs-sdk to stay lightweight
+// Axelar chains: https://github.com/axelarnetwork/axelarjs-sdk/blob/53a957deb1209325b1e3d109e0985a64db6d9901/src/constants/EvmChain.ts#L1
+export const axelarChains: Partial<Record<Chain, string>> = {
+  Ethereum: "ethereum",
+  Monad: "monad",
+  Sepolia: "ethereum-sepolia",
+  // add more as needed
+};
+
+// https://github.com/axelarnetwork/axelarjs-sdk/blob/53a957deb1209325b1e3d109e0985a64db6d9901/src/libs/TransactionRecoveryApi/AxelarRecoveryApi.ts#L16
+export enum GMPStatus {
+  SRC_GATEWAY_CALLED = "source_gateway_called",
+  DEST_GATEWAY_APPROVED = "destination_gateway_approved",
+  DEST_EXECUTED = "destination_executed",
+  EXPRESS_EXECUTED = "express_executed",
+  DEST_EXECUTE_ERROR = "error",
+  DEST_EXECUTING = "executing",
+  APPROVING = "approving",
+  FORECALLED = "forecalled",
+  FORECALLED_WITHOUT_GAS_PAID = "forecalled_without_gas_paid",
+  NOT_EXECUTED = "not_executed",
+  NOT_EXECUTED_WITHOUT_GAS_PAID = "not_executed_without_gas_paid",
+  INSUFFICIENT_FEE = "insufficient_fee",
+  UNKNOWN_ERROR = "unknown_error",
+  CANNOT_FETCH_STATUS = "cannot_fetch_status",
+  SRC_GATEWAY_CONFIRMED = "confirmed",
+}
+
+export interface GMPError {
+  txHash: string;
+  chain: string;
+  message: string;
+}
+
+export function getAxelarApiUrl(network: Network): string {
+  return network === "Mainnet"
+    ? "https://api.axelarscan.io"
+    : "https://testnet.api.axelarscan.io";
+}
+
+export function getAxelarChain(chain: Chain): string {
+  const axelarChain = axelarChains[chain];
+  if (!axelarChain) {
+    throw new Error(`Unsupported axelar chain: ${chain}`);
+  }
+  return axelarChain;
+}
+
+export async function getAxelarGasFee(
+  network: Network,
+  sourceChain: Chain,
+  destinationChain: Chain,
+  gasLimit: bigint,
+  timeoutMs = 10000
+): Promise<bigint> {
+  const url = `${getAxelarApiUrl(network)}/gmp/estimateGasFee`;
+  const axelarSourceChain = getAxelarChain(sourceChain);
+  const axelarDestinationChain = getAxelarChain(destinationChain);
+
+  const controller = new AbortController();
+  const timeoutId = setTimeout(() => controller.abort(), timeoutMs);
+
+  // Set a minimum fee of 1 to avoid 0-fee issue with relays not proceeding
+  // past the gas paid step
+  let fee = 1n;
```

Comment:
I noticed that if the fee is set to 0 (which can happen when the Axelar API incorrectly returns 0 - an issue they’re working on), the relay can get stuck on the "gas paid" step and never progress until the user tops up the gas on Axelarscan. Since no error is surfaced in the transfer status response, the simplest solution is to set the fee to 1, which guarantees that an error will appear.